internal/provider/resource_apply_module.go

```diff
@@ -554,6 +685,60 @@ func (t *applyModuleResource) getCurrentApplied(ctx context.Context,
 	return nil
 }
 
+// extractRunError extracts the error from a run's logs (if the run errored out).
+func (t *applyModuleResource) extractRunError(ctx context.Context, run *sdktypes.Run) diag.Diagnostics {
+	var diags diag.Diagnostics
+
+	if run.Status == sdktypes.RunErrored {
+		var jobID string
+
+		switch {
+		case run.Apply != nil:
+			if run.Apply.CurrentJobID != nil {
+				jobID = *run.Apply.CurrentJobID
+			}
+		case run.Plan != nil:
+			if run.Plan.CurrentJobID != nil {
+				jobID = *run.Plan.CurrentJobID
+			}
+		}
+
+		if jobID == "" {
+			diags.AddWarning("Run status is errored, but no job ID found", "")
+			return diags
+		}
+
+		logs, err := t.client.Job.GetJobLogs(ctx, &sdktypes.GetJobLogsInput{ // unlimited
+			JobID: jobID,
+		})
+		if err != nil {
+			diags.AddError("Failed to get job logs", err.Error())
+			return diags
+		}
+
+		// Find the first mention of "error" in the logs.
+		splitLogs := strings.Split(logs.Logs, "\n")
+		foundIx := -1
+		for i, log := range splitLogs {
+			if strings.HasPrefix(strings.ToLower(log), "error") {
+				foundIx = i
+				break
+			}
+		}
+
+		// Must format the message as a single string.  Otherwise, an upper layer inserts
+		// the first/outer run's error's root cause into the logs between what we print here.
+		if foundIx >= 0 {
+			diags.AddError("Error from provider-launched run:\n|\t"+
```

Comment:
The `Error from provider-launched run` and end of error string can be removed.

Reply:
Do you prefer the vertical bar and tab prefix for the lines from the inner run be printed or not?